Bottom: 86c76245825e622a785f2ad9fa1fbe45408a19c2
Top:    f5d66eb10e9cf6e99e038c41afeec88f72f751b2
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-05 14:51:13 -0700

Refresh of rs-spin

---

diff --git a/src/rsocket.c b/src/rsocket.c
index b8ae3b4..3619af4 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -955,7 +955,7 @@ static int rs_get_comp(struct rsocket *rs, int nonblock, int (*test)(struct rsoc
 		gettimeofday(&e, NULL);
 		poll_time = (e.tv_sec - s.tv_sec) * 1000000 +
 			    (e.tv_usec - s.tv_usec) + 1;
-	} while (poll_time < polling_time);
+	} while (poll_time <= polling_time);
 
 	ret = rs_process_cq(rs, 0, test);
 	return ret;
@@ -1522,7 +1522,7 @@ int rpoll(struct pollfd *fds, nfds_t nfds, int timeout)
 		gettimeofday(&e, NULL);
 		poll_time = (e.tv_sec - s.tv_sec) * 1000000 +
 			    (e.tv_usec - s.tv_usec) + 1;
-	} while (poll_time < polling_time);
+	} while (poll_time <= polling_time);
 
 	rfds = rs_fds_alloc(nfds);
 	if (!rfds)
